fix(tags): import of tag resources failed with 403 error

nlevee opened the pull request:
<!--
Please read through https://git.io/fj2m9 if this is your first time opening a
pull request to this repo. Find us in https://crossplane.slack.com
if you need any help contributing.
-->

### Description of your changes

I've changed the way tags are imported when “managementPolicies” is set to “Observe”.

Currently, the import fails with at best a 403 error. 

![image](https://github.com/user-attachments/assets/6f1e81b6-95df-4e72-a433-25542c6584d2)

The problem is that the Terraform provider's import function expects to receive the “name” as an argument, with only the name as the value (without the “tagXXX/” prefix).

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

<!--
Before reviewers can be confident in the correctness of this pull request, it
needs to tested and shown to be correct. Briefly describe the testing that has
already been done or which is planned for this change.
-->

Tested with resources in example directory.

Link to #627

[contribution process]: https://git.io/fj2m9

nlevee commented:
Hi!

Is there something wrong ? Please can you check if I have to rework something :pray:

nlevee commented:
Hi @ulucinar @sergenyalcin @turkenf,

Is there anything I can do to pass this PR ?

jeanduplessis commented:
@nlevee We're aiming for a new provider release next week and will look to get this PR in for it. 
Would you mind rebasing your PR on the latest from `main` branch since its quite a few commits behind already.

jeanduplessis commented:
/test-examples="examples/tags/v1beta1/tagkey.yaml"

jeanduplessis commented:
/test-examples="examples/tags/v1beta1/tagvalue.yaml"

nlevee commented:
> @nlevee We're aiming for a new provider release next week and will look to get this PR in for it. Would you mind rebasing your PR on the latest from `main` branch since its quite a few commits behind already.

Done 👍 thank you ☺️

turkenf reviewed (commented):
Hi @nlevee, thank you for your efforts in this PR. I left two comments for your consideration as an initial review.

turkenf reviewed (changes requested):
@nlevee, I have successfully tested the import of an existing resource with the example you gave, but when I try to create the resources from scratch, I see the following error. Can you create the resources successfully?
```
  conditions:
  - lastTransitionTime: "2024-11-01T15:05:53Z"
    message: 'observe failed: failed to observe the resource: [{0 Error when reading
      or editing TagsTagKey "key": googleapi: Error 400: com.google.apps.framework.request.StatusException:
      <eye3 title=''INVALID_ARGUMENT''/> generic::INVALID_ARGUMENT: Invalid CRM resource
      name: ''tagKeys/key''.  []}]'
    reason: ReconcileError
    status: "False"
    type: Synced
```

nlevee commented on config/tags/config.go:
```diff
@@ -19,15 +19,18 @@ func Configure(p *config.Provider) {
 			TerraformName: "google_tags_tag_value",
 			Extractor:     common.ExtractResourceIDFuncPath,
 		}
+		config.MarkAsRequired(r.TerraformResource, "parent", "tag_value")
```
Because those fileds are required in GCP API to retrieve and set.

nlevee commented:
> @nlevee, I have successfully tested the import of an existing resource with the example you gave, but when I try to create the resources from scratch, I see the following error. Can you create the resources successfully?
> 
> ```
>   conditions:
>   - lastTransitionTime: "2024-11-01T15:05:53Z"
>     message: 'observe failed: failed to observe the resource: [{0 Error when reading
>       or editing TagsTagKey "key": googleapi: Error 400: com.google.apps.framework.request.StatusException:
>       <eye3 title=''INVALID_ARGUMENT''/> generic::INVALID_ARGUMENT: Invalid CRM resource
>       name: ''tagKeys/key''.  []}]'
>     reason: ReconcileError
>     status: "False"
>     type: Synced
> ```

I'll try to reproduce on my side, I'll give you feed back asap.

turkenf commented:
/test-examples="examples/tags/v1beta1/tagkey.yaml"

turkenf commented:
I added data source for project id to the examples and removed the manual intervention, you can see the error in the [uptest run.](https://github.com/crossplane-contrib/provider-upjet-gcp/actions/runs/11720643900/job/32646503439)

![Screenshot 2024-11-07 at 13 16 28](https://github.com/user-attachments/assets/7a584501-97bf-446b-a33b-6696deba3007)

nlevee commented:
> I added data source for project id to the examples and removed the manual intervention, you can see the error in the [uptest run.](https://github.com/crossplane-contrib/provider-upjet-gcp/actions/runs/11720643900/job/32646503439)
> 
> ![Screenshot 2024-11-07 at 13 16 28](https://private-user-images.githubusercontent.com/103541666/383914097-7a584501-97bf-446b-a33b-6696deba3007.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEwNzIzMjQsIm5iZiI6MTczMTA3MjAyNCwicGF0aCI6Ii8xMDM1NDE2NjYvMzgzOTE0MDk3LTdhNTg0NTAxLTk3YmYtNDQ2Yi1hMzNiLTY2OTZkZWJhMzAwNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwOFQxMzIwMjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYzA2M2EyMDgwZmQxOGUzMzQ0MjU0MWVlOGIyZDUzZTdmNTgzNzExNDFjN2VmYTZlM2FjZWM1MTQyMjI2ZmExJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.0yjyrwH0sIkr1azGZyUYSbWRXXCVSkjWoWMCjjnKVwo)

When you create a resource, you cannot set the label `crossplane.io/external-name`, because the name is built Google API. I think that's why there is this error.

nlevee commented:
Hi @turkenf, can you help me ? :pray: 

I realy don't understand why it try to get the resources before create... The external-name is not predictable.

nlevee commented:
Hi @turkenf :wave:,
Can you help us to fix this PR ?

nlevee commented:
Hi @turkenf, do you had time to review it ?
